Update cri-dockerd switchover task for v1.24 release (#32665)
kubernetes/website: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/migrate-dockershim-dockerd.md

---
title: "Migrate Docker Engine nodes from dockershim to cri-dockerd"
weight: 9
content_type: task 
---

{{% thirdparty-content %}}

> This page shows you how to migrate your Docker Engine nodes to use `cri-dockerd`
> instead of dockershim. If you want to switch away from using dockershim and still
> use Docker Engine to run containers in Kubernetes, you can follow these steps.
> If you want to upgrade to Kubernetes v{{< skew currentVersion >}} and your
> existing cluster relies on dockershim, you will have to migrate away and
> `cri-dockerd` is one of your options.

Reviewer: ```suggestion
This page shows you how to migrate your Docker Engine nodes to use `cri-dockerd`
instead of dockershim. You should follow these steps in these scenarios:

 * You want to switch away from dockershim and still use Docker Engine to run
    containers in Kubernetes.
 * You want to upgrade to Kubernetes v{{< skew currentVersion >}} and your
    existing cluster relies on dockershim, in which case you must migrate 
    from dockershim and `cri-dockerd` is one of your options.
```

{{</note>}}

If you want to migrate to `cri-dockerd` so that you can continue using Docker
Engine as your container runtime, you should do the following for each affected
node: 

1.  Install `cri-dockerd`.
1.  Cordon and drain the node.
1.  Configure the kubelet to use `cri-dockerd`. 
1.  Restart the kubelet.
1.  Verify that the node is healthy.

Test the migration on non-critical nodes first.

You should perform the following steps for each node that you want to migrate
to `cri-dockerd`.

## {{% heading "prerequisites" %}}

*   [`cri-dockerd`](https://github.com/mirantis/cri-dockerd#build-and-install)
    installed and started on each node.
*   A [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/).

## Cordon and drain the node

1.  Cordon the node to stop new Pods scheduling on it:

    ```shell
    kubectl cordon <NODE_NAME>
    ```
    Replace `<NODE_NAME>` with the name of the node.

1.  Drain the node to safely evict running Pods: 

    ```shell
    kubectl drain <NODE_NAME> \
        --ignore-daemonsets
    ```

## Configure the kubelet to use cri-dockerd

> The following steps apply to clusters set up using the kubeadm tool. If you use

Reviewer: Should kubeadm be in backticks? I'm not sure

Reviewer 2: kubeadm should not be in backticks unless its a command
e.g. https://kubernetes.io/docs/reference/setup-tools/kubeadm/

Reviewer 3: +1 it should not be in backticks.

Author: https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names disagrees, but I'm fine with leaving it.
